pysoa/client/client.py

```diff
@@ -1009,4 +1009,5 @@ def _make_context_header(self, switches=None, correlation_id=None, context_extra
         # Add any extra stuff
         if context_extra:
             context.update(context_extra)
-        return context
+        # context keys need to be guaranteed unicode
+        return {six.text_type(k): v for k, v in six.iteritems(context)}
```

Comment:
I have concerns about this. This is going to add overhead to every PySOA request, whether it needs it or not (whereas the below only adds the overhead when needed). In case there was any misunderstanding, when I said this:

>please also fix whatever caused your client to send bytes objects as context dict keys

I wasn't referring to making a change to the PySOA client code. I was referring to fixing whatever code was using the client and sending non-unicode keys to it.

Perhaps I'm splitting hairs here and the overhead doesn't really matter. I'm not outright rejecting it, just saying that I have concerns. 🤔